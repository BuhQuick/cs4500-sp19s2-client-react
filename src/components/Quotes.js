import React from 'react'
const Quotes = ({quotes, quote,
  selectQuote, updateForm, createQuote,
  updateQuote, deleteQuote}) =>
<div>
    <h3>Quotes {quotes.length}</h3>
    <table className="table">
        <tbody>
        <tr>
            <td>
                <input
                    className="job-title-fld"
                    onChange={e => updateForm(e)}
                    value={quote.jobTitle}></input>
                <button
                    className="create-quote-btn"
                    onClick={createQuote}>Add</button>
                <button
                    className="update-quote-btn"
                    onClick={updateQuote}>Update</button>
            </td>
        </tr>
        {
            quotes
                .map(quote =>
                    <tr className="quote-row"
                        key={quote.id}>
                        <td className="job-title">
                            {quote.jobTitle}
                        </td>
                        <td>
                            <button
                                className="delete-quote-btn"
                                onClick={() => deleteQuote(quote.id)}>
                            Delete
                        </button>
                        <button
                            className="select-quote-btn"
                            onClick={() => selectQuote(quote.id)}>
                            Select
                        </button>
                        </td>
                    </tr>
                )
        }
        </tbody>
    </table>
</div>

export default Quotes
