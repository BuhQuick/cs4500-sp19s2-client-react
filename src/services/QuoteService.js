export default class QuoteService {
    static instance = null;
    static getInstance() {
        if(QuoteService.instance === null) {
            QuoteService.instance = new QuoteService()
        }
        return this.instance
    }
    findQuoteById = quoteId =>
        fetch(`http://localhost:8080/api/quotes/${quoteId}`)
            .then(response => response.json())
    findAllQuotes = () =>
        fetch("http://localhost:8080/api/quotes")
            .then(response => response.json())
    createQuote = quote => {
        delete quote.id;
        return fetch(`http://localhost:8080/api/quotes`, {
            method: 'post',
            body: JSON.stringify(quote),
            headers: {
                'content-type': 'application/json'
            }
        })
            .then(response => response.json())
    }
    updateQuote = quote =>
        fetch(`http://localhost:8080/api/quotes/${quote.id}`, {
            method: 'put',
            body: JSON.stringify(quote),
            headers: {
                'content-type': 'application/json'
            }
        }).then(response => response.json())
    deleteQuote = id =>
        fetch(`http://localhost:8080/api/quotes/${id}`, {
            method: 'delete'
        })

}
