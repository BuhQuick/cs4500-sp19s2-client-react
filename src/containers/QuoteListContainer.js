import React from 'react'

import Quotes from "../components/Quotes";
class QuoteListContainer extends React.Component {
    constructor(props) {
        super(props)
        this.quoteService = this.props.service
        this.state = {
            quotes: [],
            quote: {
                jobTitle: ''
            }
        }
    }
    componentDidMount() {
        this.findAllQuotes()
    }
    findAllQuotes = () =>
        this.quoteService
            .findAllQuotes()
            .then(quotes => {
                this.setState({
                    quotes: quotes
                })
            }
            )
    updateForm = e =>
        this.setState({
            quote: {
                id: this.state.quote.id,
                jobTitle: e.target.value
            }
        })
    createQuote = () =>
        this.quoteService
            .createQuote(this.state.quote)
            .then(this.findAllQuotes)
    deleteQuote = id =>
        this.quoteService
            .deleteQuote(id)
            .then(this.findAllQuotes)
    selectQuote = id =>
        this.quoteService
            .findQuoteById(id)
            .then(quote => this.setState({
                quote: quote
            }))
    updateQuote = () =>
        this.quoteService
            .updateQuote(this.state.quote)
            .then(this.findAllQuotes)
    render() {
        return (
            <div>
                <Quotes
                    updateQuote={this.updateQuote}
                    selectQuote={this.selectQuote}
                    deleteQuote={this.deleteQuote}
                    createQuote={this.createQuote}
                    updateForm={this.updateForm}
                    quote={this.state.quote}
                    quotes={this.state.quotes}/>
            </div>
        )
    }
}
export default QuoteListContainer