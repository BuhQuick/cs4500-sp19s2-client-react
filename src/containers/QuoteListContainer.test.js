import React from 'react'
import QuoteListContainer from './QuoteListContainer'
import Quotes from '../components/Quotes'
import TestRenderer from 'react-test-renderer';
import QuoteService from '../services/QuoteService'
const quoteService = QuoteService.getInstance()
import '../services/QuoteService.mock'

test('[QuoteListContainer renders correctly]', () => {
    const testRenderer = TestRenderer.create(
        <QuoteListContainer
            service={quoteService}/>)
    let tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()

    const testInstance   = testRenderer.root
    let   jobTitleFld    = testInstance.findByProps({className: 'job-title-fld'})

    jobTitleFld.props.onChange({target: {value: 'J123'}})
    tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()
})
