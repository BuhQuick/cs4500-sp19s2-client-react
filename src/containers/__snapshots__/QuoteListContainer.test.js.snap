// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[QuoteListContainer renders correctly] 1`] = `
<div>
  <div>
    <h3>
      Quotes 
      0
    </h3>
    <table
      className="table"
    >
      <tbody>
        <tr>
          <td>
            <input
              className="job-title-fld"
              onChange={[Function]}
              value=""
            />
            <button
              className="create-quote-btn"
              onClick={[Function]}
            >
              Add
            </button>
            <button
              className="update-quote-btn"
              onClick={[Function]}
            >
              Update
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`[QuoteListContainer renders correctly] 2`] = `
<div>
  <div>
    <h3>
      Quotes 
      0
    </h3>
    <table
      className="table"
    >
      <tbody>
        <tr>
          <td>
            <input
              className="job-title-fld"
              onChange={[Function]}
              value="J123"
            />
            <button
              className="create-quote-btn"
              onClick={[Function]}
            >
              Add
            </button>
            <button
              className="update-quote-btn"
              onClick={[Function]}
            >
              Update
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
